community/pmc-tool/main/views.py

```diff
@@ -58,7 +60,22 @@ def save_data_to_database(request):
     logging.info("Verifying issues:")
     assert_equal(issue_results, issue_authors_count)
 
+    # Find authors with the first pull request in the last week
```

Comment:
this code cannot be part of `save_data_to_database`.  These are completely different workflows.

- `save_data_to_database` runs in the background - let's say once every hour. This code is pretty slow, right? It can take a long time (several seconds to several minutes) to download all data from github. If we keep that code as part of `views`, our users will wait for this before they can see any results.

- the code in `fetch_first_...` runs every time a user opens the website. This code **already expects the database to contain the data**. Please separate the logic.